### PR TITLE
RSDK-4788: Add Readings to Python SDK

### PR DESCRIPTION
**Pull request by martha-johnston**
fix imports (from sensor to common) and use `GetReadingsRequest/Response` for movement sensor and power sensor

### CODE REVIEW

File: src/viam/components/power_sensor/power_sensor.py
Comment (martha-johnston) [inline src/viam/components/power_sensor/power_sensor.py:60]: should the below be removed? in `sensor.py` the only thing in `get_readings` is ...

File: src/viam/components/movement_sensor/movement_sensor.py
Comment (martha-johnston) [inline src/viam/components/movement_sensor/movement_sensor.py:150]: should the below be removed? in `sensor.py` the only thing in `get_readings` is `...`. Should we not be calling everything manually now that we have the request/response to use?

File: src/viam/components/movement_sensor/client.py
Comment (benjirewis) [inline src/viam/components/movement_sensor/client.py:6]: I see some import errors for these values in the test; I'm guessing `GetReadingsRe*` types were added to the `api` repo but they have yet to be automatically imported to the Python SDK.

File: src/viam/components/movement_sensor/movement_sensor.py
Comment (benjirewis) [reply src/viam/components/movement_sensor/movement_sensor.py:150]: @njooma would definitely know. My guess is that now that you're not invoking `super().get_readings()`, you only need `...` here.

Comment (randhid) [reply src/viam/components/movement_sensor/movement_sensor.py:150]: yes we should remove the bespoke implementation in each sdk, we don't need it anymore. It will now be the responsibility of the driver using these wrappers.

Comment (njooma) [reply src/viam/components/movement_sensor/movement_sensor.py:150]: This was modeled after what's done in [rdk](https://github.com/viamrobotics/rdk/blob/12d225bb4da41d580576a44e8047bb5e9e88ce20/components/movementsensor/movementsensor.go#L106) where the base class for Movement Sensor defines the `Readings` function that all subclasses can use.

File: src/viam/components/movement_sensor/client.py
Comment (njooma) [reply src/viam/components/movement_sensor/client.py:6]: Ya exactly -- this has to be merged in after the API changes get merged

File: src/viam/components/movement_sensor/movement_sensor.py
Comment (randhid) [reply src/viam/components/movement_sensor/movement_sensor.py:150]: @njooma, we added a common.proto `GetReadings` API, so we no longer have to do this in each SDK. Now we have a client and server-side implementation, and we're also changing the [go SDK](https://github.com/viamrobotics/rdk/pull/3047) implementation. So the way it's done now should be changed.

**Comment by martha-johnston** [comment]
@njooma @benjirewis @clintpurser does anyone know if I should be running make buf locally and uploading that? or should those be generated through GitHub actions? seems like test aren't passing because it doesn't have the new api methods (but it should because api was merged)

File: src/viam/services/navigation/navigation.py
Comment (benjirewis) [inline src/viam/services/navigation/navigation.py:8]: @njooma here and elsewhere we were just importing `Path` from `viam.proto.service.navigation`. I see some other API types being imported from `.`; how does that work? We added `Path` (along with `GetPathsRequest` and `GetPathsResponse`) to `src/viam/proto/service/navigation/__init__.py`, but I don't think we were able to find `Path` in `.` here.

File: tests/test_motion_service.py
Comment (benjirewis) [inline tests/test_motion_service.py:21]: I assume it's fine to just use a random camera here in the test?

File: src/viam/services/navigation/client.py
Comment (benjirewis) [inline src/viam/services/navigation/client.py:42]: [q] Why do we return API types to users? I thought we would convert the API type to some native Python class and return that to the user?

File: tests/test_motion_service.py
Comment (randhid) [reply tests/test_motion_service.py:21]: Anything that can return `GetObjectPointCluds` for an obstacle. which cameras do.

File: src/viam/services/navigation/client.py
Comment (randhid) [reply src/viam/services/navigation/client.py:42]: I want to request that follow-up API cleanup that doesn't have to do with Martha's readings change be handled in a follow-up ticket. This design does look wrong, though ) I think it should return a Python class but I don't want to make a decision given we're doing this on the fly.

**Comment by randhid** [comment]
@njooma @benjirewis @clintpurser  all good?

**Review by randhid — APPROVED** [review]

**Review by benjirewis — APPROVED** [review]
LGTM but probably wait for final approval from @njooma

File: src/viam/services/navigation/client.py
Comment (njooma) [reply src/viam/services/navigation/client.py:42]: It's not bad to return a protobuf message. It's actually preferred IMO. This way we don't have to create a bunch of classes that mirror the proto, nor we do we have to keep those classes updated.

File: src/viam/services/navigation/navigation.py
Comment (njooma) [reply src/viam/services/navigation/navigation.py:8]: The `.` syntax is only if the class is included in the `src/viam/services/navigation/__init__.py`, not the `.../proto/...` file.

File: src/viam/services/navigation/client.py
Comment (benjirewis) [reply src/viam/services/navigation/client.py:42]: @njooma that makes sense. Per recent conversation in the SDKs channel, it seems like it's just language-dependent whether the protos are "good enough" to not need native-mirrored types.

**Comment by martha-johnston** [comment]
@njooma I think I covered everything... I do want movement sensor to no longer be a subclass of sensor so that should be fixed now, I updated the docstring, and added get_readings to ExampleMovementSensor. also updated the nav __init__.py file so path could be imported along with GeoPoint, Waypoint, etc.

File: src/viam/services/navigation/__init__.py
Comment (njooma) [inline src/viam/services/navigation/__init__.py:2]: This will work, but you should also add it in the `__all__` list below

**Review by njooma — APPROVED** [review]
Almost!! One minor change. Approving it because you should be able to merge it in once you add that additional change

File: tests/test_robot.py
Comment (martha-johnston) [inline tests/test_robot.py:63]: made the __all__ change, but this was causing the tests to fail... I removed this line because they were both called "movement_sensor1" but this was of subtype "sensor", and now the tests pass. Just want to make sure this checks out @njooma

Comment (njooma) [reply tests/test_robot.py:63]: Ohh nice find. There's another spot where this should be updated, but I made a [ticket](https://viam.atlassian.net/browse/RSDK-5461) for it for myself for later (not required for this)

**Review by njooma — APPROVED** [review]

**Pull request merged**
